pulpcore/cli/ansible/repository.py

```diff
@@ -0,0 +1,146 @@
+from typing import Optional
+
+import click
+
+from pulpcore.cli.common.generic import (
+    list_entities,
+    show_by_name,
+    destroy_by_name,
+)
+from pulpcore.cli.common.context import (
+    pass_pulp_context,
+    pass_repository_context,
+    PulpContext,
+    PulpRepositoryContext,
+)
+from pulpcore.cli.ansible.context import (
+    PulpAnsibleRemoteRoleContext,
+    PulpAnsibleRemoteCollectionContext,
+    PulpAnsibleRepositoryContext,
+)
+
+
+@click.group()
+@click.option(
+    "-t",
+    "--type",
+    "repo_type",
+    type=click.Choice(["ansible"], case_sensitive=False),
+    default="ansible",
+)
+@pass_pulp_context
+@click.pass_context
+def repository(ctx: click.Context, pulp_ctx: PulpContext, repo_type: str) -> None:
+    if repo_type == "ansible":
+        ctx.obj = PulpAnsibleRepositoryContext(pulp_ctx)
+    else:
+        raise NotImplementedError()
+
+
+repository.add_command(show_by_name)
+repository.add_command(list_entities)
+repository.add_command(destroy_by_name)
+
+
+@repository.command()
+@click.option("--name", required=True)
+@click.option("--description")
+@click.option("--remote")
+@pass_repository_context
+@pass_pulp_context
+def create(
+    pulp_ctx: PulpContext,
+    repository_ctx: PulpRepositoryContext,
+    name: str,
+    description: Optional[str],
+    remote: Optional[str],
+) -> None:
+    repository = {"name": name, "description": description}
+    if remote is not None:
+        try:
+            r_href: str = PulpAnsibleRemoteRoleContext(pulp_ctx).find(name=remote)["pulp_href"]
+        except click.ClickException:
+            r_href = PulpAnsibleRemoteCollectionContext(pulp_ctx).find(name=remote)["pulp_href"]
+        repository["remote"] = r_href
+
+    result = repository_ctx.create(body=repository)
+    pulp_ctx.output_result(result)
+
+
+@repository.command()
+@click.option("--href", required=True)
+@click.option("--name")
+@click.option("--description")
+@click.option("--remote")
+@pass_repository_context
+@pass_pulp_context
+def update(
+    pulp_ctx: PulpContext,
+    repository_ctx: PulpRepositoryContext,
+    href: str,
+    name: Optional[str],
+    description: Optional[str],
+    remote: Optional[str],
+) -> None:
+    repository = repository_ctx.show(href)
+
+    if name is not None:
+        if name != "":
+            repository["name"] = name
+
+    if description is not None:
+        if description == "":
+            # unset the description
+            description = None
+        if description != repository["description"]:
+            repository["description"] = description
+
+    if remote is not None:
+        if remote == "":
+            # unset the remote
+            r_href: str = ""
```

Comment:
We are not limited to 640 kb anymore. ;)
```suggestion
            remote_href: str = ""
```